Feature module4

AlehYemelyanchyk opened the pull request:
Module 4 Spring Security for the cod review.

olegpic committed:
featureModule4 branch created

olegpic committed:
Added Spring security, authentication.

olegpic committed:
UserDAO added existsByName methods
properties - updated

olegpic committed:
featureModule4 commit

olegpic committed:
GiftCertificate - updated db table name
UserServiceImplTest - updated save method test
Controllers - updated mapping, added @PreAuthorize
WebSecurityConfig - removed .authorizeRequests().antMatchers("/api/auth/**").permitAll()

olegpic committed:
GiftCertificate, Order - changed price to BigDecimal
application.properties - set app.jwtExpirationMs= 600000 ms
Tests - updated
AuthController - code refactoring
OrderController - BigDecimal refactoring

olegpic committed:
Properties - server.port changed tp 8081

olegpic committed:
Controllers updated empty mapping
UserController - added PreAuthorize by id
AuthToken filter - jwtUtils change to getUserIdFromJwtToken
JwtUtils added getUserIdFromJwtToken method
UserDetailsServiceImpl - set to work with UserService, added loadUserById method

olegpic committed:
Code cleaning.

olegpic committed:
Migrated to SpringData
application-properties - changed the datasource to gift_certificates_dev2
cleaned pom.xml files from the redundant dependencies

aolshevskiy commented on gift-certificate-data/src/main/java/com/epam/esm/giftcertificatemodule4/entity/Role.java:
```diff
@@ -0,0 +1,78 @@
+
+@Entity
+@Table(name = "roles")
+public class Role extends BaseEntity implements Serializable {
```
This class seems to be rendundant. See https://docs.jboss.org/hibernate/core/3.6/reference/en-US/html/collections.html#collections-ofvalues

aolshevskiy commented on gift-certificate-data/src/main/java/com/epam/esm/giftcertificatemodule4/entity/User.java:
```diff
@@ -19,10 +20,24 @@
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
```
See comment to Role class

aolshevskiy commented on gift-certificate-data/src/main/java/com/epam/esm/giftcertificatemodule4/entity/User.java:
```diff
@@ -19,10 +20,24 @@
     private String name;
 
+    @Column(name = "password")
+    private String password;
```
I'd recommended to use `char[]` to store password. See https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords

aolshevskiy reviewed (approved):
Code works around Spring Security architecture. Let's discuss this offline.

olegpic committed:
GiftCertificateServiceImpl - fixed update method.
GiftCertificateServiceImpl - updated.
pom.xml files updated

olegpic committed:
UserDetailsImpl and UserDetailsServerImpl moved to services package.
application.properties updated.
Unused payload removed.
Package controller renamed to controllers
SpringSecurity rebuild from scratch.
pom.xml - jjwt-impl and jjwt-jackson added.

olegpic committed:
UserDetailsImpl - added new constructor
Controllers - @PreAuthorize updated
JwtTokenFilters updated to store the principal id
EnableGlobalMethodSecurity - @Configuration removed
